packages/runtime/src/state/modules/api-resources.ts

```diff
@@ -1,13 +1,26 @@
 import { APIResource, APIResourceType } from '../../api'
 import deepEqual from '../../utils/deepEqual'
 import { Action, ActionTypes } from '../actions'
+import { Branded } from '../../utils/branded'
 
-type State = Map<APIResourceType, Map<string, APIResource | null>>
+type CompositeResourceId = Branded<string, 'CompositeResourceId'>
+
+function compositeId(resourceId: string, locale?: string | null): CompositeResourceId {
+  return (locale != null ? `${resourceId}@${locale}` : resourceId) as CompositeResourceId
+}
+
+function parseCompositeId(compositeId: CompositeResourceId): [string, string | undefined] {
```

Comment:
Could the return type be `[string, string | null]`?

Reply:
For consistency with `compositeId` signature? We can do that, although `locale` in `compositeId` can also be `undefined`, and the intentional conversion to `null` here might make it looks like it matter which one it is, when the semantics is the same for either.

Reply:
I think I like @arvinpoddar's suggestion. To make the semantics be the same, the implementation could look like this:

```ts
  const [resourceId, locale = null] = compositeId.split('@')
  return [resourceId, locale]
```